.github/workflows/codeql-analysis.yml

```diff
@@ -17,17 +17,17 @@ jobs:
 
     steps:
       - name: Checkout Repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@v4.2.2
```

Comment:
v4 is supported so use v4 instead of 4.2.2 do this changes for all tags if provided repo has `vX.Y.Z` `vX.Y` and `vX` tag prefer `vX` > `vX.Y` > `vX.Y.Z`